Scripts/Editor/SceneViewPickerSceneGui.cs

```diff
@@ -460,6 +463,38 @@ private void FindObjectsOfTypeInSceneOrPrefab(Type type, ref List<Object> @objec
                     objects.RemoveAt(i);
             }
         }
+
+        private void FindGameObjectsInSceneOrPrefab(ref List<Object> @objects)
+        {
+            Scene currentScene = GetScene(propertyPicking.serializedObject.targetObject);
+
+            // If there is a valid scene, use that scene to find candidates instead.
+            objects.Clear();
+            if (currentScene.IsValid())
+            {
+                GameObject[] rootGameObjects = currentScene.GetRootGameObjects();
+                for (int i = 0; i < rootGameObjects.Length; i++)
+                {
+                    Transform[] transforms = rootGameObjects[i].GetComponentsInChildren<Transform>();
+                    for (int j = 0; j < transforms.Length; j++)
+                    {
+                        Transform transform = transforms[j];
+                        objects.Add(transform.gameObject);
+                    }
+                }
```

Comment:
I am not a fan of the nested loop, but wasn't sure if you would allow LINQ usage so opted for this solution anyway.